exec/cnex/lib/posix.c

```diff
@@ -0,0 +1,781 @@
+{
+    Number mode = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fchmod(number_to_sint32(fildes), number_to_sint32(mode))));
+}
+
+#if HAVE_AT_FUNCTIONS
+void posix_fchmodat(struct tagTExecutor *exec)
+{
+    Number flag = top(exec->stack)->number; pop(exec->stack);
+    Number mode = top(exec->stack)->number; pop(exec->stack);
+    char *path = string_asCString(top(exec->stack)->string); pop(exec->stack);
+    Number fd = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fchmodat(number_to_sint32(fd), path, number_to_sint32(mode), number_to_sint32(flag))));
+    free(path);
+}
+#endif
+
+void posix_fchown(struct tagTExecutor *exec)
+{
+    Number group = top(exec->stack)->number; pop(exec->stack);
+    Number owner = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fchown(number_to_sint32(fildes), number_to_sint32(owner), number_to_sint32(group))));
+}
+
+#if HAVE_AT_FUNCTIONS
+void posix_fchownat(struct tagTExecutor *exec)
+{
+    Number flag = top(exec->stack)->number; pop(exec->stack);
+    Number group = top(exec->stack)->number; pop(exec->stack);
+    Number owner = top(exec->stack)->number; pop(exec->stack);
+    char *path = string_asCString(top(exec->stack)->string); pop(exec->stack);
+    Number fd = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fchownat(number_to_sint32(fd), path, number_to_sint32(owner), number_to_sint32(group), number_to_sint32(flag))));
+    free(path);
+}
+#endif
+
+void posix_fork(struct tagTExecutor *exec)
+{
+    push(exec->stack, wrap(fork()));
+}
+
+void posix_fpathconf(struct tagTExecutor *exec)
+{
+    Number name = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fpathconf(number_to_sint32(fildes), number_to_sint32(name))));
+}
+
+void posix_fsync(struct tagTExecutor *exec)
+{
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fsync(number_to_sint32(fildes))));
+}
+
+void posix_ftruncate(struct tagTExecutor *exec)
+{
+    Number length = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(ftruncate(number_to_sint32(fildes), number_to_uint64(length))));
+}
+
+void posix_getegid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getegid())));
+}
+
+void posix_geteuid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(geteuid())));
+}
+
+void posix_getgid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getgid())));
+}
+
+void posix_gethostname(struct tagTExecutor *exec)
+{
+    char name[256];
+    Cell *r = wrap(gethostname(name, 256));
```

Comment:
Use `sizeof(name)` here instead of repeating "256".

Reply:
Fixed.